src/main/common/time.c

```diff
@@ -107,7 +118,7 @@ static void dateTimeFormat(char *buf, dateTime_t *dateTime, int16_t offset)
     int tz_hours = 0;
     int tz_minutes = 0;
 
-    if (offset != 0) {
+    if (offset != 0 && rtcIsDateTimeValid(dateTime)) {
```

Comment:
I would move the `rtcIsDateTimeValid()` check just after the `if (offset != 0)` block. We don't want to format an invalid dateTime_t even if we didn't adjust the offset.

Also, if we're not formatting the dateTime_t, it's probably a good idea to write a `\0` to `buf[0]`, to prevent a caller from passing a string with potential garbage around.

Reply:
Sounds reasonable.